lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -4,68 +4,54 @@ class Deployment < KubernetesResource
     TIMEOUT = 5.minutes
 
     def sync
-      json_data, _err, st = kubectl.run("get", type, @name, "--output=json")
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
       @found = st.success?
-      @rollout_data = {}
+      @rollout_data = { "replicas" => 0 }
       @status = nil
-      @representative_pod = nil
-      @pods = []
+      @latest_rs = nil
 
       if @found
-        @rollout_data = JSON.parse(json_data)["status"]
-          .slice("updatedReplicas", "replicas", "availableReplicas", "unavailableReplicas")
-        @status = @rollout_data.map { |replica_state, num| "#{num} #{replica_state}" }.join(", ")
+        deployment_data = JSON.parse(raw_json)
+        latest_rs_data = get_latest_rs(deployment_data)
```

Comment:
Or `find_latest_rs` since this method actually gets multiple RS and finds the right one.